packages/paste-core/components/progress-steps/src/ProgressSteps.tsx

```diff
@@ -36,16 +34,83 @@ export interface ProgressStepsProps extends Omit<HTMLPasteProps<"div">, "childre
   orientation?: Orientation;
 }
 
+const ItemSeparatortyles: {
+  [key in Orientation]: Record<string, Record<string, string | BoxStyleProps>>;
+} = {
+  vertical: {
+    "[role='listitem']>*>div:first-child::after": {
+      content: "''",
+      borderLeftWidth: "borderWidth20",
+      borderLeftStyle: "solid",
+      borderLeftColor: "colorBorderWeaker",
+      borderRadius: "borderRadius0",
+      minHeight: "sizeBase80",
+      flexGrow: "1",
+      marginY: "space30",
+    },
+
+    "[role='listitem']:last-child>*>div:first-child::after": {
+      content: "none",
+    },
+  },
+  horizontal: {
+    "[role='listitem']": {
+      display: "flex",
+      alignItems: "center",
+      flexGrow: "1",
+      "&::after, &::before": {
+        content: "''",
+        borderBottomWidth: "borderWidth20",
+        borderBottomStyle: "solid",
+        borderBottomColor: "colorBorderWeaker",
+        borderRadius: "borderRadius20",
+        minWidth: "sizeBase40",
+        flexGrow: 1,
+      },
+
+      "&>*:first-child": {
+        marginX: "space30",
+      },
+    },
+    "[role='listitem']:first-child": {
+      "&::before": {
+        content: "none",
+      },
+
+      "&>*:first-child": {
+        marginLeft: "space0",
+      },
+    },
+    "[role='listitem']:last-child": {
+      "&::after": {
+        content: "none",
+      },
+
+      "&>*:last-child": {
+        marginRight: "space0",
+      },
+    },
+  },
+};
+
 export const ProgressSteps = React.forwardRef<HTMLDivElement, ProgressStepsProps>(
   ({ element = "PROGRESS_STEPS", orientation = "horizontal", ...props }, ref) => {
+    const ContainerStyled = styled.div(
+      css({
+        ...ItemSeparatortyles[orientation],
+      }),
+    );
+
     return (
       <ProgressStepsContext.Provider value={{ orientation }}>
         <Box
           {...safelySpreadBoxProps(props)}
+          as={ContainerStyled as any}
           ref={ref}
           element={element}
-          display="flex"
           role="list"
+          display="flex"
+          {...ItemSeparatortyles[orientation]}
```

Comment:
Why are we applying the styles here if they are already set in `ContainerStyled`?